### ISSUE TITLE
Exception: no repository known with hash

### ISSUE THREAD

**Issue by joshmoore** [ome/omero-downloader#45]
I tried the latest version but I can’t get past the error below. I don’t know what ‘repository’ is in this context so I don’t know if I’m missing something, and I can’t see it in the guide (https://omero-guides.readthedocs.io/en/latest/download/docs/download.html). Is it a bug or am I just not holding it right?

Thanks,

Michael.

```
C:\Users\mporter\Downloads\OMERO.downloader-0.2.2-release\OMERO.downloader-0.2.2
>download.bat -b .\tmp\repo -s nightshade.openmicroscopy.org -u mike -w ***** -f
binary Image:4329667
cannot retrieve repositories
finding target images... done
mapping filesets of images...caught unexpected exception: java.lang.IllegalState
Exception: no repository known with hash 065ab24b-3979-46db-8853-dd8dfc9235f9
```

From: Josh Moore [mailto:notifications@github.com]
Sent: 11 November 2020 08:13
To: ome/omero-web <omero-web@noreply.github.com>
Cc: Michael Porter (Staff) <m.porter@dundee.ac.uk>; Mention <mention@noreply.github.com>
Subject: Re: [ome/omero-web] Improve user experience with large image downloads (#241)

see also: https://trello.com/c/HXHVwDXg/140-omeroweb-download-issue

@mporter-gre<https://github.com/mporter-gre>, you might also try https://github.com/ome/omero-downloader in the mean time.

_Originally posted by @mporter-gre in https://github.com/ome/omero-web/issues/241#issuecomment-725314445_

**Comment by dominikl** [comment]
Unfortunately I can't replicate the problem. Tested latest version against nightshade and merge-ci. Or could it be a file size problem? Tested only with a ~500mb czi file.

**Comment by joshmoore** [comment]
I wonder if this comes down to @mporter-gre's special status as user:1 on nightshade...

**Comment by mporter-gre** [comment]
Hi Dominik

The problem was happening over the 500 MB mark. I can supply a file if needed?

Michael.

**Comment by joshmoore** [comment]
> I can supply a file if needed?

Definitely can't hurt to retry the exact file.

**Comment by mporter-gre** [comment]
Happy to! What’s the preferred method these days?

**Comment by joshmoore** [comment]
> Happy to! What’s the preferred method these days?

Perhaps just the ID on nightshade?

Edit: Doh. I have that above. I'll give it a try.

**Comment by joshmoore** [comment]
Ha. As `root` or myself (an admin), I see:

```
finding target images... done
mapping filesets of images... done
(1/1) determining files used by image 4329667... done
(1/1, 0/0) no files to download for image 4329667
```

As `mike` (group member) or `jason` (group owner), I see the error above.

**Comment by mporter-gre** [comment]
Absolutely. The id of the first image of the set is 4392224. The dataset is 79727.

**Comment by joshmoore** [comment]
I see the same thing as myself.

**Comment by pwalczysko** [comment]
@joshmoore not sure if this is very valuable, but might exclude the "special user owning the image" suspicion.
I have downloaded the image using OMERO.web and re-imported it into ns server as myself.
Then went on to attempt to download it (now the image is owned by me and I am the downloading agent too).

Result is

```
./download.sh -b /tmp/repo -s nightshade.openmicroscopy.org -u pxxx -w xxx  -f binary Image:4471884
finding target images.... done
mapping filesets of images... done
(1/1) determining files used by image 4471884... done
(1/1, 0/0) no files to download for image 4471884
```
